File: backend/internal/model/notification.go
package model

import "time"

type Notification struct {
	ID            string    `json:"id"`
	UserID        string    `json:"user_id"`
	Type          string    `json:"type"`
	Title         string    `json:"title"`
	Message       string    `json:"message"`
	IsRead        bool      `json:"is_read"`
	ReferenceType *string   `json:"reference_type,omitempty"`
	ReferenceID   *string   `json:"reference_id,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
}

File: frontend/src/types/notification.ts
export interface NotificationItem {
  id: string;
  user_id: string;
  type: string;
  title: string;
  message: string;
  is_read: boolean;
  reference_type?: string | null;
  reference_id?: string | null;
  created_at: string;
}

export interface NotificationFilters {
  page?: number;
  perPage?: number;
  read?: boolean;
}
